### PR TITLE
Equipment category view

### PR DESCRIPTION
**Pull request by ForNeVeR**
I've started work on the inventory screen. Today we'll have only equipment category list that looks like this:

image

Yeah, that looks like a tiny change, but really it took some work to implement all this `ItemsViewModel` and `ListView` machinery. That will be reused many times for implementing other item lists in inventory / equipment screens.

### CODE REVIEW

**Comment by ForNeVeR** [comment]
cc @rexim

File: src/main/scala/ru/org/codingteam/keter/scenes/game/GameMapView.scala [hunk 1]
```diff
@@ -5,17 +5,15 @@ import ru.org.codingteam.keter.game.objects._
 import ru.org.codingteam.keter.map.TraverseUtils.{BoardCell, BoardCoords}
 import ru.org.codingteam.keter.map.{Surface, TraverseUtils}
 import ru.org.codingteam.keter.ui.IView
+import ru.org.codingteam.keter.ui.shape.Rectangle
 import ru.org.codingteam.keter.util.Logging
 import ru.org.codingteam.rotjs.interface.Display
 import ru.org.codingteam.rotjs.wrappers._
 
-class GameMapView(parent: GameScene, width: Int, height: Int, viewModel: GameMapViewModel) extends IView with Logging {
+class GameMapView(shape: Rectangle, parent: GameScene, viewModel: GameMapViewModel) extends IView with Logging {
```
Comment (rexim) [inline src/main/scala/ru/org/codingteam/keter/scenes/game/GameMapView.scala]: > `shape: Rectangle`

Why is the name of the variable more general than its type? It looks kind of strange ._.

**Comment by rexim** [comment]
Nothing critical. I approve this shit.

File: src/main/scala/ru/org/codingteam/keter/scenes/game/GameMapView.scala (hunk 1 above)
Comment (ForNeVeR) [inline src/main/scala/ru/org/codingteam/keter/scenes/game/GameMapView.scala]: Currently we support only rectangular shapes for views, but in future there maybe will be more.

Comment (rexim) [inline src/main/scala/ru/org/codingteam/keter/scenes/game/GameMapView.scala]: Ok

Comment (ForNeVeR) [inline src/main/scala/ru/org/codingteam/keter/scenes/game/GameMapView.scala]: I think now I see your point. This "shape" is not just some general name as in "geometrical shape"; it is _attribute name_ as in _shape of the view element_. So it is definitely normal.

**Pull request merged**